lib/api/unittest/CJsonOutputWriterTest.cc

```diff
@@ -1298,7 +1298,7 @@ BOOST_AUTO_TEST_CASE(testWriteCategoryDefinition) {
         ml::core::CJsonOutputStreamWrapper outputStream(sstream);
         ml::api::CJsonOutputWriter writer("job", outputStream);
 
-        writer.writeCategoryDefinition(categoryId, terms, regex, maxMatchingLength, examples);
+        writer.writeCategoryDefinition(categoryId, terms, regex, maxMatchingLength, examples, std::size_t(0), vector<int>);
```

Comment:
```suggestion
        writer.writeCategoryDefinition(categoryId, terms, regex, maxMatchingLength, examples, 0, {});
```

I wouldn't bother explicitly saying the `0` is `size_t` in this case as the function being called enforces it.  Although pedantically you're creating an `int` that's got to be widened to a `size_t` in reality the optimizer won't generate code that does that.

And for your empty `vector` you should be able to pass an `initializer_list`, i.e. `{}`.